src/perception/cost_map/cost_map.processing.cpp

```diff
@@ -0,0 +1,54 @@
+#include "cost_map.hpp"
+
+namespace mrover {
+
+    auto CostMapNodelet::pointCloudCallback(sensor_msgs::PointCloud2ConstPtr const& msg) -> void {
+        assert(msg);
+        assert(msg->height > 0);
+        assert(msg->width > 0);
+
+        if (!mPublishCostMap) return;
+
+        try {
+            SE3d zed_to_map = SE3Conversions::fromTfTree(mTfBuffer, "map", "zed_left_camera_frame");
+            auto* points = reinterpret_cast<Point const*>(msg->data.data());
+            // std::for_each(points, points + msg->width * msg->height, [&](auto* point) {
+            for (auto point = points; point - points < msg->width * msg->height; point += mDownSamplingFactor) {
+                Eigen::Vector4d p{point->x, point->y, point->z, 1};
+                point_matrix.col((point - points) / mDownSamplingFactor) = p;
+                Eigen::Vector4d normal{point->normal_x, point->normal_y, point->normal_z, 0};
+                normal_matrix.col((point - points) / mDownSamplingFactor) = normal;
+            }
+
+            // TODO(neven): Make sure this still works with manif
+            point_matrix = zed_to_map.transform() * point_matrix;
```

Comment:
Eigen should make this super fast. Good call on batch multiplying the points! We are ditching our SO3/SE3 class but I think it is in the best interest of performance. So that tradeoff has to be made.